File: new-algorithm/src/javaalgo/math/boj2292.java
package javaalgo.math;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class boj2292 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());

        int cnt = 1;
        int range = 2;

        if (N == 1) {
            System.out.println(1);
        } else {
            while (range <= N) {
                range += 6 * cnt;
                cnt++;
            }
            System.out.println(cnt);
        }
    }
}

File: new-algorithm/src/kotlinalgo/math/boj2292.kt
package kotlinalgo.math

import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val N = Integer.parseInt(readLine())

    var cnt = 1
    var range = 2

    println(when {
        N == 1 -> 1
        else -> {
            while (range <= N) {
                range += 6 * cnt
                cnt++
            }
            cnt
        }
    })
}